src/run_language_modeling.py

```diff
@@ -248,7 +248,7 @@ def run_language_modeling(model_args: ModelArguments, training_args: TrainingArg
                 weight=model_args.number_token_loss_weight
             )
 
-    if model_args.model_name_or_path:
+    if model_args.model_name_or_path and dataset_args.mode != "dataset_comparison":
```

Comment:
i think the check for dataset_comparison here is not correct, as we still want to be able to load the model weights, also when we want to do evaluation with dataset comparison